src/settings/types/AngleDegrees.h

```diff
@@ -0,0 +1,74 @@
+//Copyright (c) 2018 Ultimaker B.V.
+//CuraEngine is released under the terms of the AGPLv3 or higher.
+
+#ifndef ANGLEDEGREES_H
+#define ANGLEDEGREES_H
+
+#include <cmath> //For fmod.
+
+namespace cura
+{
+
+/*
+ * \brief Represents an angle in degrees.
+ *
+ * This is a facade. It behaves like a double, but this is using clock
+ * arithmetic which guarantees that the value is always between 0 and 360.
+ */
+struct AngleDegrees
```

Comment:
I skipped converting all of the integer angles since this angle is a double and I didn't want to risk changing the type of those variables to doubles.

Reply:
It's only a risk if you do it without checking whether it causes problems.

It is quite a bit of work to check the code for such problems, but I think you introduced the AngleDegrees class because of elegance, while now you only use it in some places, which makes the code base more messy.

I want to maintain that we should either convert all angle variables to the new type - or change none.

Reply:
The original reason why these classes were introduced was because the `Settings::get` function is overloaded with the types it should return, but needed different behaviours for some doubles than for other doubles.

Just like with `int` vs. the more specific `coord_t`, the plan is to replace occurrences of these types with their more specific types as we go. We can work through the entire code base to find all ints that are used as coordinates or dimensions with `coord_t` and all ints that are layer indices with `LayerIndex` and such, but that easily adds an additional 8 story points and additional risk to this issue. In my opinion, boy scouting is more efficient.